delta/sourcing-psql/src/test/scala/ch/epfl/bluebrain/nexus/delta/sourcing/stream/SupervisorSuite.scala

```diff
@@ -0,0 +1,307 @@
+      _ <- assertWatchRestarts(Offset.at(3L), 3, 2)
+      _ <- assertDescribe(
+             runnableByNode1,
+             PersistentSingleNode,
+             // The number of restarts has been incremented
+             restarts = 1,
+             Completed,
+             expectedProgress
+           )
+      // As it is persistent, progress has also been saved to database
+      _ <- projections.progress(runnableByNode1.name).assertSome(expectedProgress)
+      // The restart has been acknowledged and now longer comes up
+      _ <- projections.restarts(Offset.at(2L)).assertEmpty
+    } yield ()
+  }
+
+  test("Destroy a registered persistent projection") {
+    assertDestroy(runnableByNode1, Task.unit)
+  }
+
+  test("Should restart a failing projection") {
+    for {
+      _ <- assertCrash(runnableByNode1, TransientSingleNode)
+      _ <- sv.describe(runnableByNode1.name).map(_.exists(_.restarts > 0)).eventually(true)
+    } yield ()
+  }
+
+  test("Destroy a failing projection") {
+    assertDestroy(runnableByNode1, Task.unit)
+  }
+
+  test("Obtain the correct running projections") {
+    val watchRestartProgress = ProjectionProgress(Offset.at(3L), Instant.EPOCH, 3, 2, 0)
+    val runnableProgress     = ProjectionProgress(Offset.at(20L), Instant.EPOCH, 20, 0, 0)
+    for {
+      _ <- assertStart(runnableByNode1, PersistentSingleNode)
+      _ <- sv.getRunningProjections()
+             .eventually(
+               List(
+                 SupervisedDescription(
+                   metadata = Supervisor.watchRestartMetadata,
+                   EveryNode,
+                   restarts = 0,
+                   Running,
+                   progress = watchRestartProgress
+                 ),
+                 SupervisedDescription(
+                   runnableByNode1,
+                   PersistentSingleNode,
+                   0,
+                   Completed,
+                   runnableProgress
+                 )
+               )
+             )
+    } yield ()
+  }
+
+  test("Run and properly destroy a projection with an unstable destroy method") {
+    val projection = ProjectionMetadata("test", "unstable-global-projection", None, None)
+    for {
+      _               <- assertStart(projection, EveryNode)
+      // Destroy the projection with a destroy method that fails and eventually succeeds
+      unstableDestroy <- UnstableDestroy()
+      _               <- assertDestroy(projection, unstableDestroy.attempt)
+      _               <- unstableDestroy.isCompleted.assert(true, "The destroy method should have completed")
+    } yield ()
+  }
+
+  test("Run and properly destroy a projection with an failing destroy method") {
+    val projection = ProjectionMetadata("test", "unstable-global-projection", None, None)
+    for {
+      _         <- assertStart(projection, EveryNode)
+      alwaysFail = Task.raiseError(new IllegalStateException("Fail !"))
+      _         <- assertDestroy(projection, alwaysFail)
+    } yield ()
+  }
```

Comment:
This is pretty nice, but a small change like this can help make it more like given/when/then

```
  test("Run and properly destroy a projection with an unstable destroy method") {
    val projection = ProjectionMetadata("test", "unstable-global-projection", None, None)
    val destroyThatAlwaysFails = Task.raiseError(new IllegalStateException("Fail !"))

    for {
      _               <- startProjection(projection, EveryNode)
      _               <- destroyProjection(projection, destroyThatFailsThenSucceeds)
      _               <- assertProjectionIsDestroyed(projection)
    } yield ()
  }
```

I realise that technically we are asserting during the startup, but it's not necessary to communicate this, only that the start is what is happening